src/rpc/blockchain.cpp

```diff
@@ -1498,10 +1531,334 @@ UniValue reconsiderblock(const JSONRPCRequest& request)
     return NullUniValue;
 }
 
+template<typename T>
+static T CalculateTruncatedMedian(std::vector<T>& scores)
+{
+    size_t size = scores.size();
+    if (size == 0) {
+        return 0;
+    }
+
+    std::sort(scores.begin(), scores.end());
+    if (size % 2 == 0) {
+        return (scores[size / 2 - 1] + scores[size / 2]) / 2;
+    } else {
+        return scores[size / 2];
+    }
```

Comment:
For my own clarification: if there's an even number of scores, there's no single median to choose, so average the two scores before and at the midpoint. If there's an odd number of scores, there's a median value?